### PR TITLE
Af4 implement af3 redrive

### PR DESCRIPTION
**Pull request by WhiteAu**

### CODE REVIEW

File: active/af4-trashman.py [hunk 1]
```diff
@@ -68,6 +68,26 @@
...
+
+#Brat Checks
 generate_job_id >> check_brat_staleness >> report_stale_brat_jobs >> send_stale_brat_email
 generate_job_id >> check_brat_completeness >> remove_complete_brat_jobs
-[report_stale_brat_jobs, remove_complete_brat_jobs] >> mark_job_complete
\ No newline at end of file
+
+
+#Resynthesis Checks
+generate_job_id >> check_resynth_tasks >> redrive_resynth_jobs
```
Comment (esilgard) [inline active/af4-trashman.py:90]: will this mean three separate job ids?

File: active/af4-trashman.py [hunk 2]
```diff
@@ -68,6 +68,26 @@
...
+
+
+[report_stale_brat_jobs, remove_complete_brat_jobs, redrive_resynth_jobs] >> mark_job_complete
```
Comment (esilgard) [inline active/af4-trashman.py:93]: are these still listed as tasks in lines 85 and 86?

File: active/operators/trashman/check_resynth_tasks.py [hunk 3]
```diff
@@ -1,9 +1,56 @@
...
+TODO: 
+this clause DOES NOT take into account (hdcorcablobid,hdcpupdatedate) jobs which failed and then later succeeded.
+Need to account for this somehow
+'''
+REDRIVEABLE_CLAUSE = ("resynth_status = {scheduled_var} AND DATEDIFF(DAY, annotation_creation_date, GetDate()) >= {delta} "
+                     " OR resynth_status = {failure_var} ".format(scheduled_var=common_variables.JOB_RUNNING,
+                                   delta=common_variables.common_variables,
+                                   failure_var=common_variables.JOB_FAILURE))
+
+SQL_FETCH_AF3_REDRIVABLE_JOBS = ("SELECT af3_runs_id"
```
Comment (esilgard) [inline active/operators/trashman/check_resynth_tasks.py]: should we keep this closer to the naming convention we have for other sql? like target_select/source_select ?

Comment (esilgard) [reply active/operators/trashman/check_resynth_tasks.py]: or if you think this is different enough that it's it's own convention...

File: active/operators/trashman/check_resynth_tasks.py [hunk 4]
```diff
@@ -10,17 +10,17 @@
 this clause DOES NOT take into account (hdcorcablobid,hdcpupdatedate) jobs which failed and then later succeeded.
 Need to account for this somehow
 '''
-REDRIVEABLE_CLAUSE = ("resynth_status = {scheduled_var} AND DATEDIFF(DAY, annotation_creation_date, GetDate()) >= {delta} "
+REDRIVEABLE_CLAUSE = ("(resynth_status = {scheduled_var} AND DATE_PART('DAY', NOW() - annotation_creation_date) >= {delta}) "
```
Comment (esilgard) [inline active/operators/trashman/check_resynth_tasks.py:13]: oOoOoOoO....learn summin new evryday....DATE_PART

**Review by esilgard — APPROVED** [review]

File: active/af4-trashman.py (hunk 1 above)
Comment (WhiteAu) [reply active/af4-trashman.py:90]: it shouldn't. it means that generate_job_id is called once, and kicks off three separate subtrees that all use the same run_id

File: active/af4-trashman.py (hunk 2 above)
Comment (WhiteAu) [reply active/af4-trashman.py:93]: Yeah. this basically just says that 'we dont mark as complete until all of the above are finished.'

File: active/operators/trashman/check_resynth_tasks.py (hunk 3 above)
Comment (WhiteAu) [reply active/operators/trashman/check_resynth_tasks.py]: maybe? I don't really like that naming convention to begin with, tbh. it doesn't tell you much about the contents of the statement.

**Comment by WhiteAu** [comment]
> 
> 
> I haven't used exit() much...presumably this will exit from this module only? and return to the next step in the dag?

That's the intent? there isn't a good pattern for exiting early that I know of... it's either this or basically have cascading NO-OP's everywhere.

**Pull request merged**